lib/thermite/config.rb

```diff
@@ -108,22 +108,31 @@ def library_name
     end
 
     #
-    # The basename of the Rust shared library.
+    # The basename of the shared library built by Cargo.
     #
-    def shared_library
-      @shared_library ||= begin
+    def cargo_shared_library
+      @cargo_shared_library ||= begin
         filename = "#{library_name}.#{shared_ext}"
         filename = "lib#{filename}" unless Gem.win_platform?
         filename
       end
     end
 
     #
+    # The basename of the Rust shared library, as installed in the {#ruby_extension_path}.
+    #
+    def shared_library
+      @shared_library ||= "#{library_name}.so"
```

Comment:
Apparently Ruby always expects the file extension to be `.so` regardless of OS. I guess that makes it simpler.